packages/firestore/test/unit/model/field_value.test.ts

```diff
@@ -485,4 +486,57 @@ describe('FieldValue', () => {
       }
     );
   });
+
+  it('estimates size correctly for fixed sized values', () => {
+    // This test verifies that each member of a group takes up the same amount
+    // of space in memory (based on its estimated in-memory size).
+    const equalityGroups = [
+      [wrap(null), wrap(false), wrap(true)],
+      [wrap(blob(0, 1)), wrap(blob(128, 129))],
+      [wrap(NaN), wrap(Infinity), wrap(1), wrap(1.1)],
+      [wrap(new GeoPoint(0, 0)), wrap(new GeoPoint(0, 0))],
+      [wrap(Timestamp.fromMillis(100)), wrap(Timestamp.now())],
+      [
+        new fieldValue.ServerTimestampValue(Timestamp.fromMillis(100), null),
+        new fieldValue.ServerTimestampValue(Timestamp.now(), null)
+      ]
+    ];
+
+    for (const group of equalityGroups) {
+      const expectedItemSize = group[0].byteSize();
+      for (const element of group) {
+        expect(element.byteSize()).to.equal(expectedItemSize);
```

Comment:
Is it worth verifying/possible to verify that no size is zero?

I'm a little worried that these tests don't actually verify that we're aggregating the sizes of objects correctly.

Reply:
This should now be taken care of by explicitly comparing the size to a provided value. I also added more types to this test case.